File: native-with-db-server/app/src/main/java/ubb/barcoaie/lab5_ma/activities/Update/UpdateActivity.java
package ubb.barcoaie.lab5_ma.activities.Update;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;

import ubb.barcoaie.lab5_ma.R;
import ubb.barcoaie.lab5_ma.VegetableViewModel;

public class UpdateActivity extends AppCompatActivity {
    private VegetableViewModel vegetableViewModel;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_update);

        RecyclerView recyclerView = findViewById(R.id.recyclerView);
        final VegetableListAdapterUpd adapter = new VegetableListAdapterUpd(new VegetableListAdapterUpd.VegetableDiff());
        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        vegetableViewModel = new ViewModelProvider(this, new ViewModelProvider.Factory() {
            @NonNull
            @Override
            public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
                return (T) new VegetableViewModel(getApplication());
            }
        }).get(VegetableViewModel.class);
        vegetableViewModel.getAllVegetables().observe(this, adapter::submitList);
    }

    @Override
    public void onBackPressed() {
        Intent replyIntent = new Intent();
        setResult(RESULT_OK, replyIntent);
        UpdateActivity.this.finish();
    }
}

File: native/app/src/main/java/ubb/barcoaie/lab2_ma/Activities/UpdateActivity.kt
package ubb.barcoaie.lab2_ma.Activities

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import com.google.android.material.textfield.TextInputEditText
import ubb.barcoaie.lab2_ma.R
import ubb.barcoaie.lab2_ma.Repo.VegetableRepoImpl

class UpdateActivity : AppCompatActivity() {
    var repo = VegetableRepoImpl()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_update)

        val updateBtn: Button = findViewById(R.id.update_button)

        val input_name: TextInputEditText = findViewById(R.id.name2)
        val input_price: TextInputEditText = findViewById(R.id.price2)

        repo = intent.extras?.get("Repo") as VegetableRepoImpl

        updateBtn.setOnClickListener {
            val input_name_text = input_name.text.toString()
            val input_price_text = input_price.text.toString().toInt()

            repo.updateVegetable(input_name_text, input_price_text)
        }
    }

    override fun onBackPressed() {
        super.onBackPressed()
        System.out.println("pressed back")
        val intent = Intent()
        intent.putExtra("Repo", repo);
        setResult(RESULT_OK, intent)
        this.finish()
    }
}